### PR TITLE
Adding NodeJS app for asset-transfer-basic sample.

### PR DESCRIPTION
**Pull request by DereckLuo**
Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

**Description**

* Adding NodeJS application for `asset-transfer-basic` sample.

### CODE REVIEW

File: asset-transfer-basic/application-javascript/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/.editorconfig:1]: would suggest not including the editor config

File: asset-transfer-basic/application-javascript/.gitignore [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/.gitignore:1]: Worth sliming down from the default.. most is irrelevant

File: asset-transfer-basic/application-javascript/enrollAdmin.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const FabricCAServices = require('fabric-ca-client');
+const { Wallets } = require('fabric-network');
+const fs = require('fs');
+const path = require('path');
+
+async function main() {
+    try {
+        // load the network configuration
+        const ccpPath = path.resolve(__dirname, '..', '..', 'test-network', 'organizations', 'peerOrganizations', 'org1.example.com', 'connection-org1.json');
+        const ccp = JSON.parse(fs.readFileSync(ccpPath, 'utf8'));
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/enrollAdmin.js:18]: worth adding a check to ensure the file is there, and fail early.. think is created by the start network so include that in the error message.

File: asset-transfer-basic/application-javascript/enrollAdmin.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        // Create a new CA client for interacting with the CA.
+        const caInfo = ccp.certificateAuthorities['ca.org1.example.com'];
+        const caTLSCACerts = caInfo.tlsCACerts.pem;
+        const ca = new FabricCAServices(caInfo.url, { trustedRoots: caTLSCACerts, verify: false }, caInfo.caName);
+
+        // Create a new file system based wallet for managing identities.
+        const walletPath = path.join(process.cwd(), 'wallet');
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/enrollAdmin.js:26]: we've got process.cwd() and __dirname being used as the basis for paths..
where is the wallet directory in relation to the connection.json?

File: asset-transfer-basic/application-javascript/invoke.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        await gateway.connect(ccp, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+
+        // Get the network (channel) our contract is deployed to.
+        const network = await gateway.getNetwork('mychannel');
+
+        // Get the contract from the network.
+        const contract = network.getContract('fabcar');
+
+        // Submit the specified transaction.
+        // createAsset transaction - requires 6 argument, ex: ('createAsset', 'asset1', 'yellow', 'Dave', 5, 1300)
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/invoke.js:43]: would expand on the comments a little here... are these examples that you could call for example.

File: asset-transfer-basic/application-javascript/package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
+{
+    "name": "asset-transfer-basic",
+    "version": "1.0.0",
+    "description": "Asset-transfer-basic application implemented in JavaScript",
+    "engines": {
+        "node": ">=8",
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/package.json:6]: would push this up to match the latest LTS for node.. (applies to all the other package.json).

File: asset-transfer-basic/application-javascript/query.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+
+async function main() {
+    try {
+        // load the network configuration
+        const ccpPath = path.resolve(__dirname, '..', '..', 'test-network', 'organizations', 'peerOrganizations', 'org1.example.com', 'connection-org1.json');
+        const ccp = JSON.parse(fs.readFileSync(ccpPath, 'utf8'));
+
+        // Create a new file system based wallet for managing identities.
+        const walletPath = path.join(process.cwd(), 'wallet');
```
Comment (mbwhite) [inline asset-transfer-basic/application-javascript/query.js:21]: ditto earlier comment

File: asset-transfer-basic/application-javascript/.editorconfig (hunk 1 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/.editorconfig:1]: done.

File: asset-transfer-basic/application-javascript/.gitignore (hunk 2 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/.gitignore:1]: done.

File: asset-transfer-basic/application-javascript/enrollAdmin.js (hunk 4 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/enrollAdmin.js:26]: wallet directory should be located at `./wallet` with respect to connection.json. We will keep it consistent and use `__dirname` here. done.

File: asset-transfer-basic/application-javascript/invoke.js (hunk 5 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/invoke.js:43]: done.

File: asset-transfer-basic/application-javascript/package.json (hunk 6 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/package.json:6]: done.

File: asset-transfer-basic/application-javascript/query.js (hunk 7 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/query.js:21]: done.

File: asset-transfer-basic/application-javascript/enrollAdmin.js (hunk 3 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/enrollAdmin.js:18]: done.

File: asset-transfer-basic/application-javascript/invoke.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+        // Create a new gateway for connecting to our peer node.
+        const gateway = new Gateway();
+        await gateway.connect(ccp, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+
+        // Get the network (channel) our contract is deployed to.
+        const network = await gateway.getNetwork('mychannel');
+
+        // Get the contract from the network.
+        const contract = network.getContract('fabcar');
```
Comment (denyeart) [inline asset-transfer-basic/application-javascript/invoke.js:40]: Not fabcar :-)

File: asset-transfer-basic/application-javascript/query.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+        // Create a new gateway for connecting to our peer node.
+        const gateway = new Gateway();
+        await gateway.connect(ccp, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+
+        // Get the network (channel) our contract is deployed to.
+        const network = await gateway.getNetwork('mychannel');
+
+        // Get the contract from the network.
+        const contract = network.getContract('fabcar');
```
Comment (denyeart) [inline asset-transfer-basic/application-javascript/query.js:45]: Still not fabcar :-)

File: asset-transfer-basic/application-javascript/invoke.js (hunk 8 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/invoke.js:40]: this network name was hardcoded in `network.sh` script, and in order to test `asset-transfer-basic`, it was kept as `fabcar` for the network name in this PR.
Do you prefer to change the name in this PR or update it along with the new `network.sh` script once it is available?

Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/invoke.js]: done.

File: asset-transfer-basic/application-javascript/query.js (hunk 9 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-javascript/query.js]: done.

**Pull request merged**